app/src/editor/canvas/components/Ports/index.jsx

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React from 'react'
+import cx from 'classnames'
+
+import Port from './Port'
+import styles from './ports.pcss'
+
+type Props = {
+    api: any,
+    canvas: any,
+    className?: ?string,
+    module: any,
+    onPort: any,
+    onValueChange: any,
+}
+
+const Ports = ({
+    api,
+    canvas,
+    className,
+    module,
+    onPort,
+    onValueChange,
+}: Props) => {
+    const { outputs } = module
+    const inputs = module.params.concat(module.inputs)
+
+    return !!(inputs.length || outputs.length) && (
+        <div className={cx(className, styles.ports)}>
+            <div className={styles.inputs}>
+                {inputs.map((port, index) => (
+                    <Port
+                        api={api}
+                        canvas={canvas}
+                        /* eslint-disable react/no-array-index-key */
+                        key={port.id + index}
```

Comment:
this shouldn't need the index unless you're intentionally remounting when ports are reordered?

Reply:
have you seen non-unique port ids?

Reply:
Copied that piece over. Wasn't sure about the uniqueness of them ids. Will try w/o `index`.

Reply:
> Copied that piece over. 

@mondoreale ah, I think this may have been a basic protection against placeholder keys colliding with a real port's key. No longer required.